src/App.jsx

```diff
@@ -1,11 +1,127 @@
 import './App.css';
+import worldMap from './assets/world_map.png'
+import { getColorByRegion} from "./helpers/getColorByRegion.jsx";
+import { convertPopulationToMillions} from "./helpers/convertPopulation.jsx";
+import {useState} from "react";
+import axios from 'axios';
 
+//'countries' is als state variabele opgehaald om de opgehaalde landeninformatie op te slaan
+//'isFetched' is opgehaald om bij te houden of de informatie al is opgeslagen
 function App() {
 
+    // State voor Opdracht 1 - voor alle landen
+    /*
+    const [countries, setCountries] = useState([]);
+    const [isFetched, setIsFetched] = useState(false);
+    */
+
+    //State voor Opdracht 2 - zoekfunctionaliteit:
+    const [countryData, setCountryData] = useState(null);
+    const [searchTerm, setSearchTerm] = useState('');
+    const [error, setError] = useState('');
+
+    //Functie Opdracht 1 - ophalen van alle landen
+    /*
+    const fetchCountries = async () => {
+        try {
+            //Gebruik de juiste fields om regio-informatie ook op te halen
+            const response = await fetch('https://restcountries.com/v3.1/all?fields=name,population,flags,region');
+            const data = await response.json();
+
+            // Sorteren van landen op populatie (van laag naar hoog)
+            const sortedCountries = data.sort((a,b) => a.population - b.population);
+
+            setCountries(sortedCountries);
+            setIsFetched(true);
+        } catch (error) {
+            console.error('Error fetching countries:', error);
+        }
+    };
+    */
+
+    //Functie opdracht 2: gegevens ophalen van een specifiek land:
+    const fetchCountry = async (name) => {
+        try {
+            const response = await axios.get(`https://restcountries.com/v3.1/name/${name}?fullText=true`);
+            const data = response.data();
```

Comment:
data bij response.data is geen functie, maar een key binnen het object response. De ronde haakjes moet je dus weghalen hier.